Makefile

```diff
@@ -94,7 +94,8 @@ tests:
 	go test ./...
 
 vet:
-	go vet ./...
+	go tool vet --all .
+	! go tool vet --shadow . 2>&1 | grep -vE 'declaration of err shadows declaration at (proto/text.go:535|proto/text.go:570|protoc-gen-gogo/main.go:84)'
```

Comment:
Nice :)
Why not make a separate make vetshadow that is not part of the automated make all.
That why we can manually choose to run the tool to find bugs, but not pressure people into working around a tool that is still experimental.